src/Disks/ObjectStorages/MetadataStorageFromPlainRewritableObjectStorage.h

```diff
@@ -4,11 +4,31 @@
 
 #include <memory>
 #include <unordered_set>
+#include <IO/WriteHelpers.h>
 
 
 namespace DB
 {
 
+class MetadataStorageFromPlainRewritableObjectStorageTransaction final : public MetadataStorageFromPlainObjectStorageTransaction
+{
+public:
+    explicit MetadataStorageFromPlainRewritableObjectStorageTransaction(
+        MetadataStorageFromPlainObjectStorage & metadata_storage_, ObjectStoragePtr object_storage_)
+        : DB::MetadataStorageFromPlainObjectStorageTransaction(metadata_storage_, object_storage_)
+    {
+    }
+
+    void createEmptyFile(const std::string & path) override
```

Comment:
The other update operations of the disk are revertable and backed by an [Operation](https://github.com/ClickHouse/clickhouse-private/blob/master/src/Disks/ObjectStorages/MetadataStorageFromPlainObjectStorageOperations.h) type with defined `undo` rollback method.
I think it should be the same here:  `removeObjectIfExists` should be called if the transaction is being rolled back, but it's probably outside the scope of the fix – I can create a follow-up.

Reply:
Fixed in https://github.com/ClickHouse/ClickHouse/pull/72884.